homeassistant/components/matter/adapter.py

```diff
@@ -22,6 +26,73 @@
     from matter_server.client.models.node import MatterEndpoint, MatterNode
 
 
+@dataclass(frozen=True, init=False)
+class HardwareAddress(abc.ABC):
+    """Abstract representation of a hardware address."""
+
+    address: bytes
+
+    @staticmethod
+    def from_string(address: str) -> HardwareAddress:
+        """Create a HardwareAddress from a string representation."""
+        address_bytes = bytes.fromhex(re.sub("[^0-9a-fA-F]", "", address))
+        return HardwareAddress.from_bytes(address_bytes)
+
+    @staticmethod
+    def from_bytes(address: bytes) -> HardwareAddress:
+        """Create a HardwareAddress from a bytes representation."""
+        if len(address) == 6:
+            return EUI48(address)
+        if len(address) == 8:
+            return EUI64(address)
+        raise ValueError(f"Invalid hardware address length: {len(address)}")
+
+    def is_NULL(self) -> bool:
+        """Check if the address is NULL value/should not be used.
+
+        See: https://standards.ieee.org/wp-content/uploads/import/documents/tutorials/eui.pdf
+        'Values based on a zero-valued OUI [...] shall not be used as identifiers.'
+        """
+        return self.address[:3] == b"\x00\x00\x00"
+
+    def is_individual(self) -> bool:
+        """Check if the address is an individual address, not a group address."""
+        return self.address[0] & 0x01 == 0
+
+    def __str__(self) -> str:
+        """Return the string representation of the address."""
+        return ":".join(f"{byte:02x}" for byte in self.address)
+
+    def _valid_connection(self) -> bool:
+        return self.is_individual() and not self.is_NULL()
+
+    @abc.abstractmethod
+    def connection(self) -> tuple[str, str] | None:
+        """Get a connection tuple or None if the address is invalid."""
+
+
+@dataclass(frozen=True)
+class EUI48(HardwareAddress):
+    """Representation of a EUI-48 address."""
+
+    @override
+    def connection(self) -> tuple[str, str] | None:
+        if self._valid_connection():
+            return (dr.CONNECTION_NETWORK_MAC, str(self))
+        return None
+
+
+@dataclass(frozen=True)
+class EUI64(HardwareAddress):
+    """Representation of a EUI-64 address."""
+
+    @override
+    def connection(self) -> tuple[str, str] | None:
+        if self._valid_connection():
+            return (dr.CONNECTION_ZIGBEE, str(self))
+        return None
```

Comment:
This is a bit overkill and doesn't belong here in an individual integration.
From an architectural perspective I would propose this at the core level.

Reply:
I agree, that's why I added the comment to your earlier review. https://github.com/home-assistant/core/pull/121257#discussion_r1668494242

Can you tell me where such discussions about architecture best belong? It doesn't feel like a bug report or a feature request. 

In the end, it's not that important to me, I just noticed it and am bad at ignoring it. (As evident by the over-engineered code 🙂)

Reply:
Well, you could just make a draft PR against HA core and get feedback (or a merge that way) but if it needs a bit discussion first, its probably best to raise a discussion here: https://github.com/home-assistant/architecture/discussions